test/framework/src/main/java/org/elasticsearch/script/MockScriptEngine.java

```diff
@@ -177,6 +177,30 @@ public void execute() {
                 }
             };
             return context.factoryClazz.cast(factory);
+        } else if (context.instanceClazz.equals(ReindexScript.class)) {
+            ReindexScript.Factory factory = (parameters, ctx) -> new ReindexScript(parameters, ctx) {
+                @Override
+                public void execute() {
+                    final Map<String, Object> vars = new HashMap<>();
+                    vars.put("ctx", ctx);
+                    vars.put("params", parameters);
+                    vars.putAll(parameters);
+                    script.apply(vars);
+                }
+            };
+            return context.factoryClazz.cast(factory);
+        } else if (context.instanceClazz.equals(UpdateByQueryScript.class)) {
```

Comment:
We really need to figure out a better way to mock scripts, this has become unwieldy. :(

Reply:
Yeah, it's pretty nasty.